src/main/scala/scorex/core/newserialization/ByteStringReader.scala

```diff
@@ -0,0 +1,161 @@
+package scorex.core.newserialization
+
+import java.nio.ByteOrder
+import java.util.BitSet
+
+import akka.util.ByteString
+
+class ByteStringReader(byteString: ByteString) extends ScorexReader {
+
+  private var it = byteString.iterator
```

Comment:
From the ByteString sources.
```
  // Avoid `iterator` in performance sensitive code, call ops directly on ByteString instead
  override def iterator: ByteIterator = throw new UnsupportedOperationException("Method iterator is not implemented in ByteString")
```
@VictorMikheev I think we should follow this suggestion :)